command/agent/config.go

```diff
@@ -385,7 +385,9 @@ type ServerConfig struct {
 
 	// ProtocolVersion is the protocol version to speak. This must be between
 	// ProtocolVersionMin and ProtocolVersionMax.
-	ProtocolVersion int `hcl:"protocol_version"`
+	//
+	// Deprecated: This has never been used and will emit a warning if nonzero.
+	ProtocolVersion int `hcl:"protocol_version" json:"-"`
```

Comment:
Enhancement idea.  Should/could we introduce a deprecated tag that can be introspected to auto-generated deprecation warnings like you had to manually add to `command/agent.command.go`? I've had a lot of success with that kind of AOP approach on APIs in other projects.

Reply:
Perhaps, but this happens fairly rarely, and I think we may want to handle each deprecation on a case by case basis. For example `nomad job run` has a whole special `Warnings` slice where we include deprecation notifications: https://github.com/hashicorp/nomad/blob/v1.2.2/nomad/structs/structs.go#L6514